Userland/Libraries/LibJS/Runtime/RegExpPrototype.cpp

```diff
@@ -103,6 +103,88 @@ static void increment_last_index(GlobalObject& global_object, Object& regexp_obj
     regexp_object.set(vm.names.lastIndex, Value(last_index), true);
 }
 
+// 1.1.2.1 Match Records, https://tc39.es/proposal-regexp-match-indices/#sec-match-records
+struct MatchRecord {
+    MatchRecord(regex::Match const& match)
+        : is_undefined(match.view.is_null())
+    {
+        if (!is_undefined) {
+            start_index = match.global_offset;
+            end_index = start_index + match.view.length();
+        }
```

Comment:
This API is a bit strange TBH, this should just be an `Optional<Match(Record)>` instead instead of adding another field to the record for the undefined state

Reply:
Yeah, I went back and forth on how to describe this because `Vector<Optional<MatchRecord>>` felt strange too, but I can alias that if it starts to feel too wordy